src/Bitfinex.Client.Websocket/Responses/Orders/OrderConverter.cs

```diff
@@ -79,6 +79,15 @@ public static OrderStatus ParseStatus(string status)
                 case "partially filled":
                 case var s when s.Contains("partially filled"):
                     return OrderStatus.PartiallyFilled;
+                case "postonly canceled":
+                case var s when s.Contains("postonly canceled"):
```

Comment:
Merging the order status hides the reason why an order was cancelled and I need to know those reasons in my application.

Adding all the posssible statuses might make it difficult for the clients to use them. But right now its isnt supporting all the possible states. And its actually hiding data by converting it to the OrderType enum.

I see 2 options:

1) Add all the possible values to the OrderType enum.

2) Add an extra field (e.g. `OrderStatusText`) that holds the raw status so that clients can handle it however they want. And keep the enum for simple state checking. (or even move it to an extension method)

Reply:
Good points. I agree with both options. 
We need new enum types (same for "insufficient balance") and also that property `OrderStatusText` will be very handy. 

Also, I think some new property `bool IsCanceled` is a must, so the client of the library will be able to distinguish terminated orders. 
Something like 
```csharp
public bool IsCanceled => Status == OrderStatus.Undefined || 
                          Status == OrderStatus.Canceled || 
                          Status == OrderStatus.InsufficientBalance || ...
``` 

Could you please implement these features? 
Also don't forget, that bitfinex sends sometimes weird text as status, like: `PARTIALLY FILLED @ PRICE(AMOUNT)`, `CANCELED was: PARTIALLY FILLED @ PRICE(AMOUNT)`, etc.. 
For that reason we got there `case var s when s.Contains("xxx"):`

Reply:
Yea and it is breaking change, so we need to bump major version.

Reply:
sounds good 👍 

I'll implement these changes.